### DISCUSSION TITLE
Clarification needed: Virtual Relation? (Link to another record)

### DISCUSSION THREAD

**Discussion by candideu** [nocodb/nocodb#1004] [Discord Threads]
Hi! I noticed this option when creating a linking to another record, and wasn't familiar with it:
![image](https://user-images.githubusercontent.com/55474996/151078218-6030952f-9040-448e-a9e4-369ef98bdb8e.png)

What is a virtual relation? When should I be checking this option? I wasn't able to find any info about it [in the documentation](https://docs.nocodb.com/setup-and-usages/link-to-another-record/).

I've tried searching it up, but I still don't fully understand.

**Comment by o1lab** [comment]
Virtual relation is similar to has many within database but without an actual foreign key being defined.

Will move this to discssion.

**Reply by candideu** [reply]
I admit, I had to look up what a foreign key is. I guess I understand it in theory, but what is the use in this context?
When should a virtual relation be set / in which circumstances?

**Reply by o1lab** [reply]
I understand, its not necessary at most times. We will simplify this.

**Comment by RawmanGrey** [comment]
As of v.0.90.8 virtual relation became default on created relations. As I can see this option doesn't have impact on how tables behave, so as what is returned by REST API.  Please, if any, share caveats of using former default non-virtual relations.
